python/mxnet/module/executor_group.py

```diff
@@ -400,6 +414,12 @@ def forward(self, data_batch, is_train=None):
 
         """
         _load_data(data_batch, self.data_arrays, self.data_layouts)
+
+        if hasattr(data_batch, 'pad'):
```

Comment:
You can replace this with `self.cur_batch_pad = getattr(data_batch, 'pad', None)`